vars/main.yml

```diff
@@ -13,10 +13,10 @@ _satisfactory_systemd_service: "{{ satisfactory_service_name }}.service"
 _satisfactory_systemd_service_template: etc/systemd/system/satisfactory.service
 
 _satisfactory_options: >
-  {{ '-multihome=' + satisfactory_multihome if satisfactory_multihome }}
-  {{ '-ServerQueryPort=' + satisfactory_serverqueryport|string if satisfactory_serverqueryport }}
-  {{ '-BeaconPort=' + satisfactory_beaconport|string if satisfactory_beaconport }}
-  {{ '?listen -Port=' + satisfactory_port|string if satisfactory_port }}
+  {{ '-multihome=' + satisfactory_multihome if satisfactory_multihome else '' }}
```

Comment:
after reading again, I suppose only changed line is required to fix the error while the other variables have default values defined (different to `null`).

perhaps it is better to not define all of them ? this way, when not defined at playbook level, the option will not be written at all.
this can be preferable to rely on real values defined by the official satisfactory server instead on copy them here however it can also be dangerous if the developers change them in future this will cause a breaking change (and not visible during ansible deployment).

Reply:
I will test it without these options